.eslintrc.json

```diff
@@ -4,21 +4,51 @@
         "es2021": true,
         "jest": true
     },
+    "parser": "@typescript-eslint/parser",
     "extends": [
         "plugin:react/recommended",
-        "airbnb-base"
+        "plugin:@typescript-eslint/eslint-recommended",
+        "plugin:@typescript-eslint/recommended"
     ],
     "parserOptions": {
         "ecmaFeatures": {
             "jsx": true
         },
         "ecmaVersion": 12,
-        "sourceType": "module"
+        "sourceType": "module",
+        "project": "./tsconfig.json"
     },
     "plugins": [
-        "react"
+        "react",
+        "@typescript-eslint"
     ],
     "rules": {
-        "arrow-parens": "off"
+        "react/jsx-uses-react": "off",
+        "react/react-in-jsx-scope": "off",
+        "arrow-parens": "off",
+        "semi": "off",
+        "@typescript-eslint/semi": ["error"],
+        "@typescript-eslint/member-delimiter-style": ["error" ,{
+          "multiline": {
+              "delimiter": "semi",
+              "requireLast": true
+          },
+          "singleline": {
+              "delimiter": "semi",
+              "requireLast": false
+          },
+          "multilineDetection": "brackets"
+        }],
+        "indent": "off",
+        "@typescript-eslint/indent": ["error", 2, {
+          "ignoredNodes": ["TSTypeParameterInstantiation"]
+        }],
+        "@typescript-eslint/array-type": ["error", {
+            "default": "array"
+        }],
+        "@typescript-eslint/no-floating-promises": ["error"],
+        "@typescript-eslint/explicit-module-boundary-types": ["error"],
+        "quotes": "off",
+        "@typescript-eslint/quotes": ["error", "single"]
```

Comment:
This isn't strictly necessary but if we use
```json
"@typescript-eslint/quotes": ["error", "single", { "avoidEscape": true }]
```
it'd keep eslint from freaking out about using double quotes **only** if the containing string includes a single quote, some examples:

https://github.com/Imageus-OSS/client/blob/f959647ce21abd79b771f54dbe5d6209cea87af6/src/components/dashboard/JoinGroupButton.tsx#L69

https://github.com/Imageus-OSS/client/blob/f959647ce21abd79b771f54dbe5d6209cea87af6/src/pages/GroupInvitePage.tsx#L76

https://github.com/Imageus-OSS/client/blob/f959647ce21abd79b771f54dbe5d6209cea87af6/src/pages/NotFoundPage.tsx#L39